src/VisualStudio/Core/Def/Packaging/PackageInstallerServiceFactory.cs

```diff
@@ -482,6 +512,8 @@ private void OnWorkspaceChanged(object sender, WorkspaceChangeEventArgs e)
         private async Task<T?> PerformNuGetProjectServiceWorkAsync<T>(
             Func<INuGetProjectService, CancellationToken, ValueTask<T?>> doWorkAsync, CancellationToken cancellationToken)
         {
+            await ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
```

Comment:
i don't know if this is necessary.   but i was trying to preserve prior semantics.  @AArnott do you need to be on the UI thread to call .GetServiceAsync to get the IBrokeredServiceContainer?

Reply:
No. GetServiceAsync can be called from a background thread. But it's generally *not* safe to cast the result as you're doing unless you are on the main thread or know that the service object is an MTA object. In this case, it *is* an MTA so you're fine.